crates/biome_js_analyze/src/lint/nursery/use_semantic_elements.rs

```diff
@@ -0,0 +1,86 @@
+    ///
+    /// The `role` attribute is used to define the purpose of an element, but it should be used as a last resort. Using semantic elements like `<button>`, `<input>`, `<textarea>`, `<a>`, `<img>`, `<table>`, `<article>`, `<section>`, `<nav>`, `<aside>`, `<header>`, `<footer>`, `<main>`, `<figure>`, `<figcaption>`, `<details>`, `<summary>`, `<dialog>`, `<menu>`, `<menuitem>`, `<fieldset>`, `<legend>`, `<caption>`, `<colgroup>`, `<col>`, `<optgroup>`, `<option>`, `<select>`, `<datalist>`, `<output>`, `<progress>`, `<meter>`, `<time>`, `<audio>`, `<video>`, `<track>`, `<source>`, `<embed>`, `<object>`, `<param>`, `<iframe>`, `<canvas>`, `<map>`, `<area>`, `<svg>`, `<math>` are more accessible and provide better semantics.
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div role="checkbox">
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div role="img">
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <div>...</div>
+    /// <header>...</header>
+    /// <img alt="" src="image.jpg" />
+    /// ```
+    ///
+    pub UseSemanticElements {
+        version: "next",
+        name: "useSemanticElements",
+        language: "jsx",
+        sources: &[RuleSource::EslintJsxA11y("prefer-tag-over-role")],
+        recommended: true,
+    }
+}
+
+impl Rule for UseSemanticElements {
+    type Query = Aria<JsxOpeningElement>;
+    type State = AnyJsxAttribute;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let role_attributes = node.find_attribute_by_name("role").unwrap();
+
+        if let Some(attr) = role_attributes {
+            let extract_attributes = ctx.extract_attributes(&node.attributes());
+
+            let element = node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+            let element_name = element.text_trimmed();
+            let is_not_interative =
+                AriaRoles.is_not_interactive_element(element_name, extract_attributes);
+            if is_not_interative {
+                return Some(AnyJsxAttribute::from(attr));
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state.range(),
+                markup! {
+                    "The element with this role can be changed to a DOM element that already this role."
+                },
+            )
+            .footer_list(
+                markup! {
+                    "For examples and more information, see" <Hyperlink href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles">"WAI-ARIA Roles"</Hyperlink>
+                },
+                &["serif", "sans-serif", "monospace", "etc."],
+            ),
```

Comment:
@fujiyamaorange The link that I shared was an example to show you how to use the `footer_list` API, but in the array should put all the elements you were listing before

Reply:
@ematipico 
Let me confirm,
```
&["<button>", "<input>", "<textarea>", "<a>", "<img>", "<table>", "<article>", "<section>", "<nav>", "<aside>", "<header>", "<footer>", "<main>", "<figure>", "<figcaption>", "<details>", "<summary>", "<dialog>", "<menu>", "<menuitem>", "<fieldset>", "<legend>", "<caption>", "<colgroup>", "<col>", "<optgroup>", "<option>", "<select>", "<datalist>", "<output>", "<progress>", "<meter>", "<time>", "<audio>", "<video>", "<track>", "<source>", "<embed>", "<object>", "<param>", "<iframe>", "<canvas>", "<map>", "<area>", "<svg>", "<math>"]
```
This will work properly?

Reply:
Yeah, that's what you wanted, right?

Reply:
Yes! I'll commit it!